packages/perseus/src/types.ts

```diff
@@ -228,6 +228,11 @@ export type APIOptions = Readonly<{
     // keystroke caused text typed in the text area to appear in it
     // only after a good few seconds.
     editorChangeDelay?: number;
+    // This is a callback function that returns all the Widget Start Props
+    // after they have been transformed by the widget's transform function.
+    // This is useful for when we need to know how a widget has shuffled its
+    // the available choices.
+    getAllWidgetsStartProps?: (widgets: any) => any;
```

Comment:
I think it would be useful to have this `PerseusWidgetMap` type if we don't already.

```suggestion
    getAllWidgetsStartProps?: (widgets: PerseusWidgetMap) => PerseusWidgetMap;
```